internal/server/data/access_key.go

```diff
@@ -304,3 +304,12 @@ func ValidateRequestAccessKey(tx *Transaction, authnKey string) (*models.AccessK
 
 	return t, nil
 }
+
+func RemoveExpiredAccessKeys(tx WriteTxn) error {
+	query := querybuilder.New("UPDATE access_keys")
+	query.B("SET deleted_at = ?", time.Now().UTC())
+	query.B("WHERE expires_at <= ?", time.Now().UTC())
```

Comment:
Not a blocker, but I wonder if we want to add a buffer between when they expire and when we soft-delete them. So that if someone accidentally sets a short expiration time they can see the key exists but is expired, instead of being confused about why the key they created doesn't show up anymore.

The job only runs every 12 hours, so that's a bit of a buffer, but keys that expired a second ago when the job runs would still be deleted. Might be worth subtracting an hour or something to this time.